src/tokens/slice.ts

```diff
@@ -129,6 +138,41 @@ const slice = createSlice({
       loading: false,
       error: false,
     }),
+    fetchPriceHistoryStart: (
+      state: Draft<State>,
+      action: PayloadAction<{
+        tokenId: string
+        startTimestamp: number
+        endTimestamp: number
+      }>
+    ) => {
+      const { tokenId } = action.payload
+      const token = state.tokenBalances[tokenId]
+      if (token) {
+        token.historicalPricesUsd = {
+          ...token.historicalPricesUsd,
+          priceHistoryStatus: 'loading',
+        }
+      }
+    },
+    fetchPriceHistorySuccess: (
+      state: Draft<State>,
+      action: PayloadAction<{ tokenId: string; priceHistory: TokenPriceHistoryEntry[] }>
+    ) => {
+      const { tokenId, priceHistory } = action.payload
+      const token = state.tokenBalances[tokenId]
+      if (priceHistory && token?.historicalPricesUsd) {
+        token.historicalPricesUsd.priceHistory = priceHistory
+        token.historicalPricesUsd.priceHistoryStatus = 'success'
+      }
+    },
+    fetchPriceHistoryFailure: (state: Draft<State>, action: PayloadAction<{ tokenId: string }>) => {
+      const { tokenId } = action.payload
+      const token = state.tokenBalances[tokenId]
+      if (token?.historicalPricesUsd) {
```

Comment:
same here, seems like we might want to store the error even if there isn't existing state for that token